sermant-plugins/sermant-mq-grayscale/sermant-mq-grayscale-plugin/src/main/java/io/sermant/mq/grayscale/utils/MqGrayscaleConfigUtils.java

```diff
@@ -0,0 +1,294 @@
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.mq.grayscale.utils;
+
+import io.sermant.core.config.ConfigManager;
+import io.sermant.core.plugin.config.ServiceMeta;
+import io.sermant.core.service.dynamicconfig.common.DynamicConfigEventType;
+import io.sermant.core.utils.tag.TrafficUtils;
+import io.sermant.mq.grayscale.config.GrayTagItem;
+import io.sermant.mq.grayscale.config.MqGrayscaleConfig;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.rocketmq.common.message.Message;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * grayscale config util
+ *
+ * @author chengyouling
+ * @since 2024-06-03
+ */
+public class MqGrayscaleConfigUtils {
+    /**
+     * serviceMeta info
+     */
+    public static final Map<String, String> MICRO_SERVICE_PROPERTIES = new HashMap<>();
+
+    /**
+     * auto consumerType
+     */
+    public static final String CONSUME_TYPE_AUTO = "auto";
+
+    /**
+     * config cache key
+     */
+    private static final String CACHE_CONFIG_KEY = "mqGrayConfig";
+
+    private static final Map<String, MqGrayscaleConfig> CACHE_CONFIG = new ConcurrentHashMap<>();
+
+    private static final long DEFAULT_AUTO_CHECK_DELAY_TIME = 15L;
+
+    /**
+     * all traffic tags that's been set, using for sql92 expression reset
+     */
+    private static final Set<String> GRAY_TAGS_SET = new HashSet<>();
+
+    static {
+        ServiceMeta serviceMeta = ConfigManager.getConfig(ServiceMeta.class);
+        MICRO_SERVICE_PROPERTIES.put("version", serviceMeta.getVersion());
+        if (serviceMeta.getParameters() != null) {
+            MICRO_SERVICE_PROPERTIES.putAll(serviceMeta.getParameters());
+        }
+        CACHE_CONFIG.put(CACHE_CONFIG_KEY, new MqGrayscaleConfig());
+    }
+
+    private MqGrayscaleConfigUtils() {
+    }
+
+    /**
+     * compare mqGrayscaleConfig with trafficTags, serviceMeta, return match grayGroupTag
+     *
+     * @return grayGroupTag
+     */
+    public static String getGrayGroupTag() {
+        if (!CACHE_CONFIG.get(CACHE_CONFIG_KEY).isEnabled()) {
+            return "";
+        }
+        MqGrayscaleConfig mqGrayscaleConfig = CACHE_CONFIG.get(CACHE_CONFIG_KEY);
+        if (mqGrayscaleConfig == null) {
+            return "";
+        }
```

Comment:
Check whether the value is null and then check whether the switch is enabled.

Reply:
static{} has set value, will not null, and logic has changed